### ISSUE TITLE
clj-kondo crashes with a RuntimeException while linting a particular file which uses clara-rules 0.22.1

### ISSUE THREAD

**Issue by kieran-owens-dividend** [clj-kondo/clj-kondo#2165]
This is only reproducible using the current version of clara-rules as a dependency, so I'm not sure it's completely a kondo issue and may be related to how clara's kondo hooks are implemented. On the other hand a RuntimeException wasn't expected. 

Also, unrelated to kondo itself, and although my testing used cli kondo, the error causes the clojure-lsp server in VS Code to crash early and renders much of its code navigation features broken or unusable

**version**

clj-kondo v2023.07.13

**platform**

Native clj-kondo on OSX x64

**problem**

clj-kondo crashes with a java runtime exception: 

```
Unexpected error. Please report an issue.
java.lang.RuntimeException: java.lang.Exception: Not supported: class clojure.core$vec
	at com.cognitect.transit.impl.WriterFactory$1.write(WriterFactory.java:65)
	at cognitect.transit$write.invokeStatic(transit.clj:171)
	at clj_kondo.impl.cache$to_cache.invokeStatic(cache.clj:62)
	at clj_kondo.impl.cache$update_defs$fn__3111.invoke(cache.clj:133)
	at clojure.core$fn__8454$fn__8456.invoke(core.clj:6845)
	at clojure.core.protocols$iter_reduce.invokeStatic(protocols.clj:49)
	at clojure.core.protocols$fn__8164.invokeStatic(protocols.clj:75)
	at clojure.core.protocols$fn__8164.invoke(protocols.clj:75) .....
```

**repro**

```
(ns docs-rules
  (:require
   #_{:clj-kondo/ignore [:refer-all]}
   [clara.rules :refer :all]))

(defrecord DocRequirementInfo [capital-provider-id document-type-code product-type type required])

#_{:clj-kondo/ignore [:unresolved-symbol]}
(defquery requirements-qry
  []
  [?requirement <- DocRequirementInfo])
```

and running `clj-kondo --lint <file.clj>` demonstrates the error

**config**

Only started to happen when using the current version of clara-rules (with hooks). 

{:deps {com.cerner/clara-rules {:mvn/version "0.22.1"}}

**expected behavior**

Complain if it must, but not to crash with a RuntimeException

**Comment by borkdude** [comment]
Thanks, I'll take a look soon

**Comment by borkdude** [comment]
I found the issue, working on a fix.

**Comment by kieran-owens-dividend** [comment]
Thanks. Appreciate all the effort on this, and generally with the product.

**Comment by borkdude** [comment]
Should be fixed on master
